src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -2783,39 +2779,34 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 	private _runTaskCommand(arg?: any): void {
 		const identifier = this._getTaskIdentifier(arg);
 		const type = arg && typeof arg !== 'string' && 'type' in arg ? arg.type : undefined;
-		let task = arg && typeof arg !== 'string' && 'task' in arg ? arg.task : arg === 'string' ? arg : undefined;
-		if (identifier) {
-			this._getGroupedTasks({ task, type }).then(async (grouped) => {
+		const task = arg && typeof arg !== 'string' && 'task' in arg ? arg.task : arg === 'string' ? arg : undefined;
+		this._getGroupedTasks().then(async (grouped) => {
+			const tasks = grouped.all();
+			if (identifier) {
```

Comment:
Why did identifier move here? It should be checked before `_getGroupedTasks`, no?